semgrep-core/src/engine/Match_rules.ml

```diff
@@ -902,7 +910,19 @@ and match_range_with_pattern env r mvar opt_lang formula =
                 lazy_content
                 (Some { r with r = mval_range }) ) )
 
-and lazy_ast_of_string lang str warn_msg =
+and lazy_ast_of_string lang base_tok str warn_msg =
+  let fix_toks ast =
```

Comment:
Could use a comment about why this is necessary

Reply:
Sure thing!